### PR TITLE
Configure timeouts

### PR DESCRIPTION
**Pull request by dogukanteber**
This PR configures timeouts for the distributor and the query-frontend.

### CODE REVIEW

**Review by friedrichg — COMMENTED** [review]
As discussed offline. Let's implement these limits for now:

http_server_read_timeout
http_server_write_timeout
http_server_idle_timeout 
http_client_timeout
http_client_dialer_timeout
http_client_tls_handshake_timeout
http_client_response_header_timeout

File: gateway/config.go [hunk 1]
```diff
@@ -2,27 +2,33 @@ package gateway
...
+	Distributor   Upstream     `yaml:"distributor"`
+	QueryFrontend Upstream     `yaml:"frontend"`
+}
+
+type Upstream struct {
+	URL                             string        `yaml:"url"`
+	Paths                           []string      `yaml:"paths"`
+	HTTPClientDialerTimeout         time.Duration `yaml:"http_client_dialer_timeout"`
+	HTTPClientTLSHandshakeTimeout   time.Duration `yaml:"http_client_tls_handshake_timeout"`
+	HTTPClientResponseHeaderTimeout time.Duration `yaml:"http_client_response_header_timeout"`
```
Comment (dogukanteber) [inline gateway/config.go:23]: As mentioned in [this comment](https://github.com/cortexproject/auth-gateway/pull/18#pullrequestreview-1399450604), we should also implement `http_client_timeout`. However, I was not sure about it. In this [blog](https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/) under **Client Timeouts** section, the image shows that the supported values ie. `http_client_dialer_timeout, http_client_tls_handshake_timeout, http_client_response_header_timeout`, cover almost same as `http.Client.Timeout`. I feel like the combination of these three is the same as `http.Client.Timeout`. Therefore, I hesitated to implement it.

Can I get your opinion on this @friedrichg?

Comment (friedrichg) [reply gateway/config.go:23]: it's not exactly the same, we need to give folks the possibility to configure them all as needed. please include them

File: gateway/testdata/configuration.yaml [hunk 2]
```diff
@@ -1,6 +1,14 @@
 server:
   address: localhost
   port: 8080
+  http_server_read_timeout: 20
```
Comment (friedrichg) [inline gateway/testdata/configuration.yaml:4]:
```suggestion
  http_server_read_timeout: 20s
```

all the timeouts should be all human readable. A number is not enough. Also `1m` should be supported.

Use [time.ParseDuration](https://pkg.go.dev/time#ParseDuration)

**Review by friedrichg — COMMENTED** [review]
last nits

**Review by friedrichg — APPROVED** [review]
LGTM. Good job

**Pull request merged**